drivers/serial/uart_emul.c

```diff
@@ -168,6 +285,20 @@ uint32_t uart_emul_flush_tx_data(const struct device *dev)
 	return count;
 }
 
+static int uart_emul_init(const struct device *dev)
+{
+	ARG_UNUSED(dev);
```

Comment:
Just for my own curiosity: The compilers that I tried with did not complain at all. Is this a common accepted pattern in zephyr to avoid a ton of macro branching?